core/connectors/publishers/firehose_publisher.py

```diff
@@ -0,0 +1,68 @@
+import json
+import logging
+import time
+from http import HTTPStatus
+
+from boto3.session import Session
+
+from core.connectors.publishers.publisher_interface import PublisherInterface
+from core.models.payload_model import PayloadModel
+
+
+# from Fire
+
+
+class FirehosePublisher(PublisherInterface):
+
+    def __init__(self, params: dict):
+        self._retry_count = 0
+        self._method_response = HTTPStatus.SEE_OTHER, str(None)
+        self.logger = logging.getLogger("uvicorn.info")
+        self._aws_region = params["aws_region"]
+        self._firehose_client = Session(region_name=self._aws_region).client("firehose")
+        self._max_retries = params.get("max_retries", 3)
+
+    async def start(self):
+        pass
+
+    async def stop(self):
+        pass
+
+    async def send(self, destination: str, payload: PayloadModel) -> (int, str):
+
+        self._retry_count = 0
+        while True:
```

Comment:
in async calls, we shouldn't use blocking while block, if event is not delivered better to send the response immediately and its producer responsibility to retry